pytorch_toolkit/object_detection/tests/common/test_case.py

```diff
@@ -32,10 +34,21 @@ def get_dependencies(template_file):
         return output
 
 
-def get_epochs(template_file):
+def get_epochs_for_noncompressed_train(template_file):
     with open(template_file) as read_file:
-        content = yaml.load(read_file, yaml.SafeLoader)
-    return content['hyper_parameters']['basic']['epochs']
+        content = yaml.safe_load(read_file)
+    val_from_template = content.get('hyper_parameters', {}).get('basic', {}).get('epochs')
+    if val_from_template is not None:
+        return val_from_template
+
+    # Note that the following gets total_epochs parameter for the
+    # original config files pointed by template, whereas
+    # NNCFConfigTransformer (and may be other transformers)
+    # may change the parameters from the config file.
+    config_file_name = content['config']
+    config_path = os.path.join(os.path.dirname(template_file), config_file_name)
+    config = mmcv.Config.fromfile(config_path)
+    return config['total_epochs']
```

Comment:
2 @Ilya-Krylov 

Ilya, if we want to use compression, we cannot use `epoch` and `base_learning_rate` from the template, since we should change the values from the `model.py` -- for usual training and for compressionthe values should be different!

As an example, if for usual training the base LR is 0.05, for int8 quantization it may be 0.00025, for sparsity it may be 0.0025   
For usual training the number of epochs should be 70, for int8 quantization the number of epochs should be 2, for sparsity the number of epochs may be 50 (or some other value, @AlexKoff88 may comment it)

So, we cannot set the default values from the template.yaml.
User will be able to add them later -- if he/she wants to check the own strategy of compression (more or less epochs, etc)

But if we always write the values in template.yaml, we will have to write for each model a documentation:   
*we recommend to use 70 epochs for this model for usual training, 2 epochs for int8 quantization, 50 epochs for sparsity....*

Reply:
There should be 2 separate scripts: train.py (does not use compression and use hyper_parameters from template.yaml) and compress.py (does use compression and use its own parameters). The compress.py script should also have parameters that overrides values stored template.yaml (int8, prunning, sparsity , True/False).

But do not remove hyperparameters from template.yaml

Reply:
Ok

Reply:
Done